webapp/channels/src/packages/mattermost-redux/src/utils/data_loader.ts

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * A DataLoader is an object that can be used to batch requests for fetching objects from the server for performance
+ * reasons.
+ */
+abstract class DataLoader<Identifier, Result = unknown> {
+    protected readonly fetchBatch: (identifiers: Identifier[]) => Result;
+    private readonly maxBatchSize: number;
+
+    protected readonly pendingIdentifiers = new Set<Identifier>();
+
+    constructor(args: {
+        fetchBatch: (identifiers: Identifier[]) => Result;
+        maxBatchSize: number;
+    }) {
+        this.fetchBatch = args.fetchBatch;
+        this.maxBatchSize = args.maxBatchSize;
+    }
+
+    public queueForLoading(identifiersToLoad: Identifier[]): void {
+        for (const identifier of identifiersToLoad) {
+            if (!identifier) {
```

Comment:
I don't think that's likely. We don't have anything specific in place to prevent a valid ID from being empty, but I have a feeling that there's a lot of logic that'd break down if an ID is empty. That said, I would be fine making callers responsible for passing valid identifiers.

@M-ZubairAhmed Any thoughts on this? I think I copied this part of the code from your code

Reply:
While under heavy load we observed that there is network request send without either null or empty status id. To counter that it was added before

Reply:
We should probably look into what the root cause of that is at some point, but given we don't expect a zero or empty value to be a valid case, I'm going to leave this as-is